platform/flash/zephyr/src/mender-flash.c

```diff
@@ -40,8 +40,8 @@ mender_flash_open(char *name, size_t size, void **handle) {
     }
 
     /* Begin deployment with sequential writes */
-    if ((result = flash_img_init((struct flash_img_context *)*handle)) < 0) {
-        mender_log_error("flash_img_init failed (%d)", result);
+    if ((result = flash_img_init((struct flash_img_context *)*handle)) != 0) {
```

Comment:
Might as well do `(0 != (result = ...))` like we do in other places with other values.